### PR TITLE
Worldmap tileset expansion

### PR DESCRIPTION
**Pull request by Rusty-Box**
This PR will add several new tilesets for worldmaps and overall improvements on exising ones.
Additionally, the halloween tileset, along side a few new christmas themed tiles have been added to `ice_world.strf` as well.

**New additions are:**
- Underground Forest
- Forest to Corrupted Forest transition tiles
- Crystal decorations
- Haunted Library
- More decorations for the entire Forest theme
- Ice cliffs

**Improved tiles are:**
- The entire Snow tileset (improved tiling)
- Ice cave (better, detailed design)
- Forest and Corrupted Forest tiles (colour tweaks)
- Completely redrawn Forest and Corrupted Forest trees
- Snowy Trees
- Tower

### CODE REVIEW

**Comment by Rusty-Box** [comment]
Before merging there is yet one tiny thing I need to fix beforehand!

When magnification is not set to 100% white pixels appear on some of the forest themed sprites (decorations in particular).

**Comment by serano01** [comment]
I wonder if the old "world.strf" could be deleted after merging. It was an choice between ugly and new/polished, anyway. Only Worldmap still using it is the Halloween one.

**Comment by Rusty-Box** [comment]
@serano01 Semphris and I decided against this, for now. Since you weren't there:

TL;DR - After v0.7 is released we plan to freeze development and clean up a lot of messy design choices and flaws the game has accumulated over the years. So stuff like the old `worldmap.strf` getting replaced will be done then.

For more insight, see this [wiki page](https://github.com/SuperTux/wiki/blob/master/Cleanup-List.md).

> Only Worldmap still using it is the Halloween one.

I changed that in this PR so no worldmap uses it anymore. All have been set to `ice_world.strf`

**Comment by weluvgoatz** [comment]
> Before merging there is yet one tiny thing I need to fix beforehand!

> When magnification is not set to 100% white pixels appear on some of the forest themed sprites (decorations in particular).

This is interesting, as I see it too in-game yet when I open the files in any graphics program and magnify them I don't see the white pixels.
